packages/coverage-istanbul/src/provider.ts

```diff
@@ -105,6 +105,11 @@ export class IstanbulCoverageProvider implements CoverageProvider {
       return map
     }, {})
 
+    if (!this.ctx.isFirstRun && !this.ctx.config.coverage.watchAll && this.ctx.changedTests.size > 0) {
+      const changedTests = Array.from(this.ctx.changedTests)
```

Comment:
The `changedTests` cannot be used for resolving changed files, as changed tests can import all kinds of files. Instead, if we really want to include only the changed files, we'll need to see if Vite exposes names of the changed files in HMR-api, or something similar.

Reply:
Thank you for your feedback. I look into more options for this.